cicecore/cicedynB/dynamics/ice_dyn_shared.F90

```diff
@@ -1986,23 +1980,63 @@ subroutine viscous_coeffs_and_rep_pressure_T (strength, tinyarea, &
 
       ! NOTE: for comp. efficiency 2 x zeta and 2 x eta are used in the code
 
-!      if (trim(yield_curve) == 'ellipse') then
-
-      if (capping) then
-         tmpcalc = strength/max(Delta,tinyarea)
-      else
-         tmpcalc = strength/(Delta + tinyarea)
-      endif
-
+      tmpcalc =     capping *(strength/max(Delta,tinyarea))+ &
+                (c1-capping)*(strength/(Delta + tinyarea))
       zetax2 = (c1+Ktens)*tmpcalc
       rep_prs = (c1-Ktens)*tmpcalc*Delta
       etax2 = epp2i*zetax2
 
-!      else
+       end subroutine viscous_coeffs_and_rep_pressure_T
 
-!      endif
 
-       end subroutine viscous_coeffs_and_rep_pressure_T
+      subroutine viscous_coeffs_and_rep_pressure_T2U (zetax2T_00, zetax2T_01, &
+                                                      zetax2T_11, zetax2T_10, &
+                                                       etax2T_00,  etax2T_01, &
+                                                       etax2T_11,  etax2T_10, & 
+                                                        maskT_00,  maskT_01, &
+                                                        maskT_11,  maskT_10, &
+                                                        tarea_00,   tarea_01, &
+                                                        tarea_11,   tarea_10, &
+                                                        deltaU,               &
+                                                        zetax2U, etax2U,      &
+                                                        rep_prsU)
+                              
+
+      real (kind=dbl_kind), intent(in):: &
+        zetax2T_00,zetax2T_10,zetax2T_11,zetax2T_01, &
+         etax2T_00, etax2T_10, etax2T_11, etax2T_01,  & ! 2 x visous coeffs, replacement pressure
+         maskT_00, maskT_10, maskT_11, maskT_01, &
+         tarea_00, tarea_10, tarea_11, tarea_01, &
```

Comment:
please comment the code to explain what 00, 10, 01, 11 are

Reply:
will do. These are i+0,j+0,i+1 and j+1